synapse/storage/search.py

```diff
@@ -683,6 +693,17 @@ def f(txn):
 def _to_postgres_options(options_dict):
     return "'%s'" % (",".join("%s=%s" % (k, v) for k, v in options_dict.items()),)
 
+def _extract_search_keywords(entry):
+    """
+    Extract an iterator of words from the string.
+
+    For instance: https://github.com/matrix-org/synapse/issues/3024
+    Becomes: ["https","github","com","matrix-org","synapse","issues","3024"]
+
+    :param entry: A string to be pre-processed for searching.
+    :return: List of words in the string.
+    """
+    return re.finditer(r"([\w\-]+)", entry.value, re.UNICODE)
```

Comment:
re.UNICODE is a no-op:

```suggestion
    return re.finditer(r"([\w\-]+)", entry.value)
```